src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/RecordComponentNumberCheck.java

```diff
@@ -156,11 +154,15 @@ public void visitToken(DetailAST ast) {
      * @return the number of record components in this record definition
      */
     private static int countComponents(DetailAST recordComponents) {
-        final AtomicInteger count = new AtomicInteger(0);
-        TokenUtil.forEachChild(recordComponents,
-            TokenTypes.RECORD_COMPONENT_DEF,
-            node -> count.getAndIncrement());
-        return count.get();
+        int count = 0;
+        DetailAST child = recordComponents.getFirstChild();
+        while (child != null) {
+            if (child.getType() == TokenTypes.RECORD_COMPONENT_DEF) {
+                count++;
+            }
+            child = child.getNextSibling();
+        }
```

Comment:
```suggestion
        for (DetailAST child = recordComponents.getFirstChild();
             child != null;
             child = child.getNextSibling()) {
            if (child.getType() == TokenTypes.RECORD_COMPONENT_DEF) {
                count++;
            }
        }
```
this is the simpler fit for what we are trying to do and what actually happens.

Reply:
Okay